synapse/handlers/deactivate_account.py

```diff
@@ -121,6 +127,15 @@ async def deactivate_account(
 
         # Mark the user as erased, if they asked for that
         if erase_data:
+            # Remove avatar URL from this user
+            await self._profile_handler.set_avatar_url(
+                UserID.from_string(user_id), requester, "", by_admin=True
```

Comment:
I suspect that `by_admin` should only be true if the user ID also matches the requester's user ID?

Reply:
When `by_admin` is `false`, it is not possibile to delete the avatar url.

https://github.com/matrix-org/synapse/blob/f737368a26bb9eea401fcc3a5bdd7e0b59e91f09/synapse/handlers/profile.py#L269-L277

Under circumstances the switch should be renamed.

Which cases do we have?

- requester != target_user => deactivation by admin
- requester != target_user => deactivation by unauthorized not admin user, this should not be checked here
- requester == target_user => deactivation by myself

Reply:
I'm not sure I follow. Isn't it possible if `by_admin` is `True` or if the target user is the requester?

I don't think setting `by_admin` to `True` here makes sense since `DeactiveAccountHandler.deactivate_account` is called from user APIs as well as admin APIs.

Reply:
I made a decision for a simpler solution. I have added the `by_admin` flag to the `DeactiveAccountHandler`. So it is clear that the request comes from an admin from the Admin API.